golpa.go

```diff
@@ -404,3 +407,45 @@ func (model *Model) SolveWithContext(ctx context.Context) (res *SolveResult, err
 
 	return ret, err
 }
+
+// WriteLPFile dumps the model to the specified file in lp format.
+func (model *Model) WriteLPFile(filename string) error {
+	model.mu.RLock()
+	defer model.mu.RUnlock()
+
+	ret := C.write_lp(model.prob, C.CString(filename))
+
+	if ret != C.TRUE {
+		return fmt.Errorf("model not written successfully")
+	}
+
+	return nil
+}
+
+//export lpexCallback
+func lpexCallback(modelPtr unsafe.Pointer, buf *C.char) C.int {
+	model, ok := loadRef(modelPtr).(*Model)
+	if !ok {
+		return 0
+	}
+
+	model.writeBuf.WriteString(C.GoString(buf))
+
+	return 0
+}
+
+// WriteLP returns the model in lp format.
+func (model *Model) WriteLP() (string, error) {
```

Comment:
nitpicking: since we're not really writing anything, can't we call this `DumpLP` or `ExportLP`?

Reply:
I agree that this name is weird, but I wanted to stay as close as possible to the lpsolve naming. I think this is helping the user.

Reply:
I guess you don't agree? Should I change it to `ExportLP`?

Reply:
oops, sorry, went under the radar. Yeah, the current API isn't really tied to lpsolve (switched from glpk to lpsolve in the past; might switch again in the future). So I'd rather have something the "reads" well, than mimic lpsolve.

Reply:
I see - changed it to ExportLP.